MFC.md

```diff
@@ -7,33 +7,89 @@ is `freebsd` as suggested in other docs.
+
+   where the first 10 digits of the hash (might be longer in a large repository)
+   is used to mark the commit message. This "abbreviated hash" can be retrieved by:
+
 ```
 git show --format=%p --no-patch $full_hash
 ```
-This preserves the information,
-but isn't 'git standard' which is just to have a line of text that says the commit was from hash
-whatever without the 'key: value' structure. It also wouldn't help the `git log --cherry` command, unless
-we would use that to come up with candidates, and then use the MFC lines to weed them out.
-Alternatively, adding a note either to the src (saying where it had been merged to) or the destination
-(saying where it had been merged from) would allow for easier parsing and after-the-fact fixups
-similar to svn merge --record-only.
+
+   This preserves the information, but isn't 'git standard'.  It also
+   requires committers to manually edit commit messages to include
+   this information when merging.
+
+2. Use the `-x` flag with `git cherry-pick`.  This adds a line to the
+   commit message that includes the hash of the original commit when
+   merging.  Since it is added by Git directly, committers do not have
+   to manually edit the commit log when merging.
+
+We feel that the second option is simpler going forward.
+
+### Finding Eligible Merges
+
+One feature some developers have found happy with subversion is
+determining which commits have or have not been merged (for example,
+`svn mergeinfo --show-revs eligible`).
+
+Git provides some built-in support for this via the `git cherry` and
+`git log --cherry` commands.  These commands compare the raw diffs of
+commits (but not other metadata such as log messages) to determine if
+two commits are identical.  This works well when each commit from head
+is merged as a single commit to a stable branch, but it falls over if
+multiple commits from main are squashed together as a single commit to
+a stable branch.
+
+There are a few options for resolving this:
+
+1. We could ban squashing of commits and instead require that committers
+   stage all of the fixup / follow-up commits to stable into a single
+   push.
+
+2. We could adopt a consistent style for describing merges and write
+   our own tooling to wrap around `git cherry` to determine the list
+   of eligible commits.  A simple approach here might be to use the
+   syntax from `git cherry-pick -x`, but require that a squashed
+   commit list all of the hashes (one line per hash) at the end of
+   the commit message.  Developers could do this by using
+   `git cherry-pick -x` of each individual commit into a branch and
+   then use `git rebase` to squash the commits down into a single
+   commit, but collecting the `-x` annotations at the end of the
+   merged commit log.
+
+XXXJHB: I'm not sure what this paragraph means?
 
 There are also advantages to all the commits on stable/X having something to identify them as
```

Comment:
So this is the paragraph I wasn't sure about as I wasn't sure if you were thinking of using merge commits as an alternative and if so I needed to capture that as a 3rd option.  If that's not the case and this is just saying we want to have some kind of annotation for merges, I think I can just drop this paragraph.